kernel/rtlil.cc

```diff
@@ -483,6 +483,58 @@ pool<string> RTLIL::AttrObject::get_strpool_attribute(const RTLIL::IdString &id)
 	return data;
 }
 
+void RTLIL::AttrObject::set_strlist_attribute(const RTLIL::IdString& id, const std::vector<string> &data)
+{
+	string attrval;
+	for (const auto &s : data) {
+		if (!attrval.empty())
+			attrval += "|";
+		attrval += s;
+	}
+	set_string_attribute(id, attrval);
+}
+
+void RTLIL::AttrObject::add_strlist_attribute(const RTLIL::IdString& id, const vector<string> &data)
+{
+	string attrval = get_string_attribute(id);
+	for (const auto &s : data) {
+		if (!attrval.empty())
+			attrval += "|";
+		attrval += s;
+	}
+	set_string_attribute(id, attrval);
+}
+
+vector<string> RTLIL::AttrObject::get_strlist_attribute(const RTLIL::IdString &id) const
+{
+	return split_tokens(get_string_attribute(id), "|");
+}
+
+void RTLIL::AttrObject::append_src_attribute(const RTLIL::AttrObject &other, bool prepend)
```

Comment:
I haven't tested this helper yet. This one and the helpers above, which can be used to manipulate `src` as an ordered list, we might want to introduce in an effort to establish a defined ordered for the `src` entries.

Reply:
(This really doesn't have a good reason to be in this changeset, I am going to remove this.)

Reply:
>  This one and the helpers above, which can be used to manipulate src as an ordered list, we might want to introduce in an effort to establish a defined order for the src entries.

This would be my concern for merging this PR, yes. Right now src is considered a "pool" attribute, right?

Reply:
Yes, that’s how the codebase treats the attribute now.

We were considering to change this, and I have written those helpers, but I removed them from this PR. I made it use the usual strpool method instead.